sklbench/benchs/sklearn_estimator.py

```diff
@@ -0,0 +1,405 @@
+        'inference': get_subset_metrics_of_estimator(
+            task, 'inference', estimator_instance, x_test, y_test)
+    }
+    for method in metrics.keys():
+        for stage in estimator_methods.keys():
+            if method in estimator_methods[stage]:
+                metrics[method].update(quality_metrics[stage])
+
+    return metrics, estimator_instance
+
+
+def main(bench_case: BenchCase, filters: List[BenchCase]):
+    # get estimator class and ML task
+    library_name = get_bench_case_value(bench_case, 'algorithm:library')
+    estimator_name = get_bench_case_value(bench_case, 'algorithm:estimator')
+
+    estimator_class = get_estimator(library_name, estimator_name)
+    task = estimator_to_task(estimator_name)
+
+    # load and transform data
+    data, data_description = load_data(bench_case)
+    (x_train, x_test, y_train, y_test), data_description = split_and_transform_data(
+        bench_case, data, data_description)
+
+    # assign special values
+    assign_case_special_values_on_run(
+        bench_case, x_train, y_train, x_test, y_test, data_description)
+
+    # get estimator parameters
+    estimator_params = get_bench_case_value(bench_case, 'algorithm:estimator_params', dict())
+
+    # get estimator methods for measurement
+    estimator_methods = get_estimator_methods(bench_case)
+    online_inference_mode = get_bench_case_value(
+        bench_case, 'algorithm:online_inference_mode', False)
+
+    # benchmark case filtering
+    if not bench_case_filter(bench_case, filters):
+        logger.warning('Benchmarking case was filtered.')
+        print('[]')
+        return
+
+    # run estimator methods
+    context_class, context_params = get_context(bench_case)
+    with context_class(**context_params):
+        metrics, estimator_instance = measure_sklearn_estimator(
+            bench_case, task, estimator_class, estimator_methods, estimator_params,
+            x_train, x_test, y_train, y_test, online_inference_mode)
+
+    result_template = {
+        'task': task,
+        'dataset': get_data_name(bench_case, shortened=True),
+        'library': library_name,
+        'estimator': estimator_name,
+        'device': get_bench_case_value(bench_case, 'algorithm:device')
+    }
+    if 'assume_finite' in context_params:
+        result_template['assume_finite'] = context_params['assume_finite']
+    estimator_params = estimator_instance.get_params()
+    logger.debug(f'Estimator parameters:\n{custom_format(estimator_params)}')
+    result_template.update(estimator_params)
+
+    data_descs = {
+        'training': data_description['x_train'],
+        'inference': data_description['x_test']
+    }
+    data_descs['inference'].update({'online_inference_mode': online_inference_mode})
+    if 'n_classes' in data_description:
+        data_descs['training'].update({'n_classes': data_description['n_classes']})
+        data_descs['inference'].update({'n_classes': data_description['n_classes']})
+
+    results = []

```

Comment:
Catch should impact here, either by 1) not adding results or 2) adding specific results which signify a failure

Reply:
Same as for comment above.